eng/scripts/get-aspire-cli.ps1

```diff
@@ -0,0 +1,437 @@
+            return $ENV:PROCESSOR_ARCHITECTURE
+        }
+    }
+
+    # For PowerShell 6+ on Unix systems, use .NET runtime information
+    if ($PSVersionTable.PSVersion.Major -ge 6) {
+        try {
+            $runtimeArch = [System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture
+            switch ($runtimeArch) {
+                'X64' { return "x64" }
+                'X86' { return "x86" }
+                'Arm64' { return "arm64" }
+                default {
+                    Say-Verbose "Unknown runtime architecture: $runtimeArch"
+                    # Fall back to uname if available
+                    if (Get-Command uname -ErrorAction SilentlyContinue) {
+                        $unameArch = & uname -m
+                        switch ($unameArch) {
+                            { $_ -in @('x86_64', 'amd64') } { return "x64" }
+                            { $_ -in @('aarch64', 'arm64') } { return "arm64" }
+                            { $_ -in @('i386', 'i686') } { return "x86" }
+                            default {
+                                Say-Verbose "Unknown uname architecture: $unameArch"
+                                return "x64"  # Default fallback
+                            }
+                        }
+                    }
+                    return "x64"  # Default fallback
+                }
+            }
+        }
+        catch {
+            Write-Warning "Failed to get runtime architecture: $($_.Exception.Message)"
+            # Final fallback - assume x64
+            return "x64"
+        }
+    }
+
+    # Final fallback for older PowerShell versions
+    return "x64"
+}
+
+# taken from dotnet-install.ps1
+function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
+    Say-Verbose $MyInvocation
+
+    if ($Architecture -eq "<auto>") {
+        $Architecture = Get-Machine-Architecture
+    }
+
+    switch ($Architecture.ToLowerInvariant()) {
+        { ($_ -eq "amd64") -or ($_ -eq "x64") } { return "x64" }
+        { $_ -eq "x86" } { return "x86" }
+        { $_ -eq "arm" } { return "arm" }
+        { $_ -eq "arm64" } { return "arm64" }
+        default { throw "Architecture '$Architecture' not supported. If you think this is a bug, report it at https://github.com/dotnet/install-scripts/issues" }
+    }
+}
+
+# Common function for web requests with centralized configuration
+function Invoke-SecureWebRequest {
+    param(
+        [string]$Uri,
+        [string]$OutFile,
+        [int]$TimeoutSec = 60,
+        [int]$OperationTimeoutSec = 30,
+        [string]$UserAgent = $Script:UserAgent,
+        [int]$MaxRetries = 5
+    )
+
+    try {
+        if ($PSVersionTable.PSVersion.Major -ge 6) {
+            Invoke-WebRequest -Uri $Uri -OutFile $OutFile -SslProtocol Tls12,Tls13 -MaximumRedirection 10 -TimeoutSec $TimeoutSec -OperationTimeoutSeconds $OperationTimeoutSec -UserAgent $UserAgent -MaximumRetryCount $MaxRetries
```

Comment:
The dotnet-install.ps1 has a comment saying:

> # HttpClient is used vs Invoke-WebRequest in order to support Nano Server which doesn't support the Invoke-WebRequest cmdlet.

Do we care about Nano Server?

Reply:
cc @joperezr @davidfowl